lib/go-tc/stats_summary.go

```diff
@@ -187,3 +187,115 @@ type StatsSummaryLastUpdatedAPIResponse struct {
 	Response StatsSummaryLastUpdated `json:"response"`
 	Alerts
 }
+
+// StatsSummaryV5 is an alias for the latest minor version for the major version 5.
+type StatsSummaryV5 StatsSummaryV51
+
+// StatsSummaryV51 is a summary of some kind of statistic for a CDN and/or
+// Delivery Service.
+type StatsSummaryV51 struct {
+	CDNName         *string    `json:"cdnName"  db:"cdn_name"`
+	DeliveryService *string    `json:"deliveryServiceName"  db:"deliveryservice_name"`
+	StatName        *string    `json:"statName"  db:"stat_name"`
+	StatValue       *float64   `json:"statValue"  db:"stat_value"`
+	SummaryTime     time.Time  `json:"summaryTime"  db:"summary_time"`
+	StatDate        *time.Time `json:"statDate"  db:"stat_date"`
```

Comment:
Looks to me like the `StatsSummary` and the `StatsSummaryV51` structs are identical. The only change happens in the `parseTime` method which converts the timestamp into rfc3339 format. If that's actually the case, you don't need a new struct for it. You can just call the `parseTimeV5` method if the api version being used is >= 5.0.

Reply:
MarshalJson is where the depreciated TimeLayout is used and that is addressed through new struct StatsSummaryV5.

```
func (ss StatsSummaryV5) MarshalJSON() ([]byte, error) {
	type Alias StatsSummaryV5
	resp := struct {
		StatDate    *string `json:"statDate"`
		SummaryTime string  `json:"summaryTime"`
		Alias
	}{
		SummaryTime: ss.SummaryTime.Format(time.RFC3339),
		Alias:       (Alias)(ss),
	}
	if ss.StatDate != nil {
		resp.StatDate = util.Ptr(ss.StatDate.Format(dateFormat))
	}
	return json.Marshal(&resp)
}
```